client/common/client.c

```diff
@@ -215,6 +215,46 @@ static BOOL freerdp_client_settings_post_process(rdpSettings* settings)
 		settings->Fullscreen = TRUE;
 	}
 
+	/* deal with the smartcard / smartcard logon stuff */
+	if (settings->SmartcardEmulation)
+	{
+		/* if no pin is defined on the smartcard emulation use the user password */
+		if (!settings->SmartcardPin)
+		{
+			if (!settings->Password)
+			{
+				WLog_ERR(TAG, "No pin or password defined for smartcard emu");
+				goto out_error;
+			}
+
+			if (!freerdp_settings_set_string(settings, FreeRDP_SmartcardPin, settings->Password))
+			{
+				WLog_ERR(TAG, "error when setting smartcard pin to user password");
+				goto out_error;
+			}
+		}
+	}
+
+	if (settings->SmartcardLogon)
+	{
+		settings->TlsSecurity = TRUE;
```

Comment:
would always use `freerdp_settings_set_bool()`

Reply:
I find that `freerdp_settings_set_string()` makes sense as it simplifies the code, I can't see the benefit with `freerdp_settings_set_bool()`: it makes it super complicated for setting a bool, for no gain.

Reply:
@hardening well, not on a first glance, but I´ve done some tests internally where I make the settings struct opaque and can add debug / assert / check code in these setters.